web/src/components/Transactions/TransactionsContext.jsx

```diff
@@ -0,0 +1,85 @@
+import * as fcl from "@onflow/fcl"
+import PropTypes from "prop-types"
+import {createContext, useEffect, useRef, useState} from "react"
+import {ADD, REMOVE, RESET, UPDATE} from "./transactionsReducer"
+import usePersistedTransactionsReducer from "./usePersistedTransactionsReducer"
+import {isErrored, isSealed} from "./utils"
+
+export const TransactionsContext = createContext()
+
+export const TransactionsContextProvider = ({children}) => {
+  const subscribedTransactionsRef = useRef({})
+  const [currentUser, setCurrentUser] = useState(false)
+
+  // Subscribe to transactions on state hydration and when new transactions are added
+  const subscribeToTransaction = tx => {
+    subscribedTransactionsRef.current[tx.id] = {
+      unsub: fcl.tx(tx.id).subscribe(newData =>
+        dispatch({
+          type: UPDATE,
+          payload: {id: tx.id, data: newData},
+        })
+      ),
+    }
+  }
+
+  const onHydrate = txs => txs.forEach(subscribeToTransaction)
+  const {state, dispatch} = usePersistedTransactionsReducer(
+    currentUser,
+    onHydrate
```

Comment:
Could this be moved to the top of the file to avoid confusion about where `dispatch` is created?

Reply:
This uses the onHydrate/subscribeToTransaction functions above, so it has to come after them. subscribeToTransaction uses dispatch in a callback